src/TLuaInterpreter.cpp

```diff
@@ -16145,18 +16207,22 @@ std::pair<int, QString> TLuaInterpreter::setScriptCode(QString& name, const QStr
         return {-1, qsl("cannot have an empty string as name")};
     }
 
-    auto ids = mpHost->getScriptUnit()->findScriptId(name);
-    TScript* pS = mpHost->getScriptUnit()->getScript(ids.value(pos, -1));
```

Comment:
`pos` is not decremented here but it is in the new code, this might be the problem?

Reply:
Sort of - it is related to what I described above in https://github.com/Mudlet/Mudlet/pull/6742#discussion_r1204843174 ...